### PR TITLE
Use FS module for modlog

### PR DESCRIPTION
**Pull request by xfix**

### CODE REVIEW

**Comment by asgdf** [comment]
While ripgrepmodlog turned out to be faster for non-trivial queries anyway, I'd like to point out that replacing `checkRoomModlog()` with a simple "read the whole file" would have a rather negative impact on the performance of queries without a search term (i.e. "give me the last n lines of this file") because it requires reading the whole file. 
Probably no big deal for most rooms, but lobby and staff probably have rather large modlogs so idk about them.

**Comment by xfix** [comment]
@asgdf 

Thanks for concern. The old code to be fair was a bit on buggy side, considering it was reading 1024 bytes blocks, but there is no guarantee that it wouldn't stop reading in middle of a line.

I don't think this is that big of a deal because ripgrep instead of Node.js fallback should be used if performance actually mattered.

... actually, thinking about it, using `readline` Node.js module may be a good idea.

**Comment by asgdf** [comment]
>considering it was reading 1024 bytes blocks, but there is no guarantee that it wouldn't stop reading in middle of a line.

Lines cut in half by blocks were handled correctly, as far as I know. The leading fragment is stored and then appended to the end of the next block, resulting in the line being complete again. It should even be able to handle lines that are longer than the block size.

File: chat-plugins/modlog.js [hunk 1]
```diff
@@ -165,46 +165,21 @@ function runModlog(room, searchString, exactSearch, maxLines) {
...
-		if (startPos > 0) newLines.splice(-1, 1);
-		for (let i = 0; i < newLines.length; i++) {
-			if (newLines[i] && regex.test(newLines[i])) {
-				const insertionSuccessful = results.tryInsert(newLines[i]);
-				if (!insertionSuccessful) {
-					break outerLoop;
-				}
-			}
+async function checkRoomModlog(path, regex, results) {
+	const fileContents = await FS(path).read();
```
Comment (Zarel) [inline chat-plugins/modlog.js]: `.read()` is for the most part unnecessary.

`.readTextIfExists()` does the same thing, but automatically does the string interpretation and doesn't crash if the file doesn't exist.

Comment (Zarel) [inline chat-plugins/modlog.js]: (By which I mean it returns `""` if the file doesn't exist)

File: chat-plugins/modlog.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ async function runModlog(room, searchString, exactSearch, maxLines) {
 }
 
 async function checkRoomModlog(path, regex, results) {
-	const fileContents = await FS(path).read();
+	const fileContents = await FS(path).readTextIfExists();
 	for (const line of fileContents.toString().split('\n').reverse()) {
```
Comment (Zarel) [inline chat-plugins/modlog.js]: You don't need `toString()` anymore.

File: chat-plugins/modlog.js [hunk 3]
```diff
@@ -129,11 +129,11 @@ function checkRipgrepAvailability() {
 	return Config.ripgrepmodlog;
 }
 
-function runModlog(room, searchString, exactSearch, maxLines) {
+async function runModlog(room, searchString, exactSearch, maxLines) {
 	const useRipgrep = checkRipgrepAvailability();
 	let fileNameList = [];
 	if (room === 'all') {
-		const fileList = fs.readdirSync(`${__dirname}/${LOG_PATH}`);
+		const fileList = await FS(`${__dirname}/${LOG_PATH}`).readdir();
```
Comment (Zarel) [inline chat-plugins/modlog.js]: ...actually one of the benefits `FS` brings is that you don't need to do the whole `__dirname` thing anymore.

**Comment by Zarel** [comment]
readline is not particularly efficient, anyway. Writing it yourself was probably a better idea overall.

**Comment by Zarel** [comment]
I'm merging because, as xfix points out, we already support `ripgrep` for performance.

**Comment by Zarel** [comment]
*grumble* xfix making me manually merge stuff

**Pull request merged**